x-pack/plugins/observability_solution/entity_manager/server/lib/entities/install_entity_definition.ts

```diff
@@ -46,134 +52,135 @@ export interface InstallDefinitionParams {
   logger: Logger;
 }
 
+/**
+ * install an entity definition and all its required components after
+ * validating that the definition id is valid and available.
+ * attempt to remove all installed components if the installation fails.
+ */
 export async function installEntityDefinition({
   esClient,
   soClient,
   definition,
   logger,
 }: InstallDefinitionParams): Promise<EntityDefinition> {
-  const installState = {
-    ingestPipelines: {
-      history: false,
-      latest: false,
-    },
-    transforms: {
-      history: false,
-      backfill: false,
-      latest: false,
-    },
-    definition: false,
-    indexTemplates: {
-      history: false,
-      latest: false,
-    },
-  };
```

Comment:
this might be a controversial change but I've removed the state tracking to rollback because 
- it adds boilerplate and makes it harder to read/maintain
- all the revert operations we run in the catch block ignores 404 so the api calls will successfully execute regardless of the components being installed or not. this is not a performance masterpiece but
- I don't think the non happy case where this is useful has to be optimized